pytorch_pfn_extras/profiler/_record.py

```diff
@@ -43,13 +44,41 @@ def complete(self) -> None:
         pass
 
 
+@contextmanager
+def dummy_tracer(name: str) -> Generator[None, None, None]:
+    yield None
+
+
+@contextmanager
+def tracer(
+    tag: str,
+    device: "DeviceLike" = "cpu",
+    trace: Union[_tracing.Tracer, bool] = False,
+) -> Generator[None, None, None]:
+    # this uses the PyTorch autograd tracer or one for custom devices
+    runtime_cls = runtime_registry.get_runtime_class_for_device_spec(device)
+    runtime_tracer = runtime_cls.trace
+
+    user_tracer: _tracing.Tracer
+    if isinstance(trace, bool) and not trace:
+        user_tracer = _tracing.DummyTracer()
+    elif isinstance(trace, bool):
+        user_tracer = _tracing.get_tracer()
+    elif isinstance(trace, _tracing.Tracer):
+        user_tracer = trace
```

Comment:
If the trace object is customized by a user, this could potentially result in a non-thread-local object. Is this considered expected behavior? It might be beneficial to pass the Tracer class and then instantiate it via get_tracer.

Reply:
Yeah, the user can have a tracer he/she manages explicitly like enabling or disabling it, or flush it when needed. So the final user should be in control of the object itself. I think it is ok if its not stored in thread local. We just used the TLS for dealing with the global object.  But maybe a single global variable is fine?